### PR TITLE
MM-15847 migrating getFlaggedPosts to sync by default

### PR DESCRIPTION
**Pull request by carmo-evan**
#### Summary
migrating getFlaggedPosts to sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10981

### CODE REVIEW

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -179,22 +179,20 @@ func (s *SqlPostStore) Overwrite(post *model.Post) (*model.Post, *model.AppError
...
-			result.Err = model.NewAppError("SqlPostStore.GetFlaggedPosts", "store.sql_post.get_flagged_posts.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			for _, post := range posts {
-				pl.AddPost(post)
-				pl.AddOrder(post.Id)
-			}
+	var posts []*model.Post
+	if _, err := s.GetReplica().Select(&posts, "SELECT * FROM Posts WHERE Id IN (SELECT Name FROM Preferences WHERE UserId = :UserId AND Category = :Category) AND DeleteAt = 0 ORDER BY CreateAt DESC LIMIT :Limit OFFSET :Offset", map[string]interface{}{"UserId": userId, "Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Offset": offset, "Limit": limit}); err != nil {
+		return nil, model.NewAppError("SqlPostStore.GetFlaggedPosts", "store.sql_post.get_flagged_posts.app_error", nil, err.Error(), http.StatusInternalServerError)
+	} else {
```
Comment (jespino) [inline store/sqlstore/post_store.go]: You can remove the else here beause the if clause always returns.

Comment (hanzei) [inline store/sqlstore/post_store.go]: Nit: You can drop the `else` here.

File: store/storetest/post_store.go [hunk 2]
```diff
@@ -1374,7 +1374,7 @@ func testPostStoreGetFlaggedPostsForTeam(t *testing.T, ss store.Store, s SqlSupp
 	o5 = (<-ss.Post().Save(o5)).Data.(*model.Post)
 	time.Sleep(2 * time.Millisecond)
 
-	r1 := (<-ss.Post().GetFlaggedPosts(o1.ChannelId, 0, 2)).Data.(*model.PostList)
+	r1, _ := ss.Post().GetFlaggedPosts(o1.ChannelId, 0, 2)
```
Comment (hanzei) [inline store/storetest/post_store.go]: Please see my comment in your other PR: https://github.com/mattermost/mattermost-server/pull/10996#discussion_r288390445

File: store/storetest/post_store.go [hunk 3]
```diff
@@ -1550,7 +1552,8 @@ func testPostStoreGetFlaggedPosts(t *testing.T, ss store.Store) {
 	err := ss.Preference().Save(&preferences)
```
Comment (hanzei) [inline store/storetest/post_store.go]: Merging `master` has caused the build to fail. You can fix this by changing this:
```suggestion
	err = ss.Preference().Save(&preferences)
```

**Pull request merged**